frontend/discourse/app/components/user-autocomplete-results.gjs

```diff
@@ -0,0 +1,128 @@
+import Component from "@glimmer/component";
+import { tracked } from "@glimmer/tracking";
+import { fn } from "@ember/helper";
+import { on } from "@ember/modifier";
+import { action } from "@ember/object";
+import didInsert from "@ember/render-modifiers/modifiers/did-insert";
+import didUpdate from "@ember/render-modifiers/modifiers/did-update";
+import { and, eq, not } from "truth-helpers";
+import avatar from "discourse/helpers/avatar";
+import icon from "discourse/helpers/d-icon";
+import { safeInvoke } from "discourse/lib/function-utils";
+import { formatUsername } from "discourse/lib/utilities";
+import scrollIntoView from "discourse/modifiers/scroll-into-view";
+
+/**
+ * Component for rendering user autocomplete results for the DAutocomplete modifier.
+ *
+ * This component handles rendering of users, emails, and groups in the autocomplete
+ * dropdown, and is designed to be used with DAutocomplete's `component` API.
+ *
+ * @component UserAutocompleteResults
+ * @param {Array} results - Array of autocomplete results (users, emails, groups)
+ * @param {number} selectedIndex - Currently selected index in the results list
+ * @param {Function} onSelect - Callback function triggered when a result is selected
+ * @param {Function} onRender - Optional callback function triggered after component renders
+ */
+export default class UserAutocompleteResults extends Component {
+  static TRIGGER_KEY = "@";
+
+  @tracked isInitialRender = true;
+
+  @action
+  handleResultClick(result, index, event) {
+    event.preventDefault();
+    event.stopPropagation();
+    safeInvoke(this.args.onSelect, result, index, event);
```

Comment:
What errors might be thrown here? I think it's best for us to handle them specifically rather than adding a helper to catch & ignore all kinds of errors.